### PR TITLE
Add link to node reference from guide

### PR DESCRIPTION
**Pull request by DavidMansolino**
https://www.cyberbotics.com/doc/guide/index?version=enhancement-node-link

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:eyes:

File: guide/controller-programming.md [hunk 1]
```diff
@@ -347,7 +347,7 @@ int main() {
 ### Using Controller Arguments
 
 In the ".wbt" file, it is possible to specify arguments that are passed to a controller when it starts.
-They are specified in the `controllerArgs` field of the `Robot, Supervisor` or `DifferentialWheels` node, and they are passed as parameters of the `main` function.
+They are specified in the `controllerArgs` field of the [Robot](../reference/robot.md), [Supervisor](../reference/supervisor.md) or [DifferentialWheels](../reference/differentialwheels.md) node, and they are passed as parameters of the `main` function.
```
Comment (stefaniapedrazzi) [inline guide/controller-programming.md]: Given that DifferentialWheels is deprecated, I would maybe remove it from the generic documentation.

File: guide/epuck.md [hunk 2]
```diff
@@ -5,7 +5,7 @@
 E-puck is a miniature mobile robot originally developed at EPFL for teaching purposes by the designers of the successful Khepera robot.
 The hardware and software of e-puck is fully open source, providing low level access to every electronic device and offering unlimited extension possibilities.
 
-The model includes support for the differential wheel motors (encoders are also simulated, as position sensors), the infra-red sensors for proximity and light measurements, the accelerometer, the camera, the 8 surrounding LEDs, the body and front LEDs, bluetooth communication (modeled using Emitter / Receiver devices) and ground sensors extension.
+The model includes support for the differential wheel motors (encoders are also simulated, as position sensors), the infra-red sensors for proximity and light measurements, the accelerometer, the camera, the 8 surrounding [LEDs](../reference/led.md), the body and front [LEDs](../reference/led.md), bluetooth communication (modeled using [Emitter](../reference/emitter.md) / [Receiver](../reference/receiver.md) devices) and ground sensors extension.
```
Comment (stefaniapedrazzi) [inline guide/epuck.md]: it seems that also Camera and Accelerometer should link to the corresponding documentation

File: guide/modeling.md [hunk 3]
```diff
@@ -118,28 +118,28 @@ You will find the description about the ODE functions on [this page](http://ode-
...
 It is more accurate to specify the mass if it is known.
 If you are modeling a real robot it is sometimes possible to find the mass values in the robot's specifications.
-If you specify the densities, Webots will use the volume of each `boundingObject` multiplied by the density of the corresponding `Physics` node to compute each mass.
+If you specify the densities, Webots will use the volume of each `boundingObject` multiplied by the density of the corresponding [Physics](../reference/physics.md) node to compute each mass.
 This may be less accurate because `boundingObject`s are often rough approximations.
 
 ### How to Get a Realisitc and Efficient Rendering?
 
-The quality of the rendering depends on the `Shapes` resolution, on the setup of the `Materials` and on the setup of the `Lights`.
+The quality of the rendering depends on the [Shapes](../reference/shape.md) resolution, on the setup of the `Materials` and on the setup of the `Lights`.
```
Comment (stefaniapedrazzi) [inline guide/modeling.md]: Material and Light doesn't link to the corresponding documentation

File: guide/programming.md [hunk 4]
```diff
@@ -193,13 +193,13 @@ There are different techniques depending on what you want to draw:
...
-Please lookup for the `Supervisor` node in the `Reference Manual`.
-2. If you want to add a small sub-window in front of the 3d graphics, you should consider using the `Display` node.
+Please lookup for the [Supervisor](../reference/supervisor.md) node in the `Reference Manual`.
+2. If you want to add a small sub-window in front of the 3d graphics, you should consider using the [Display](../reference/display.md) node.
 This will allow you to do 2d vector graphics and text.
 This is also useful for example to display processed camera images.
-Please lookup for the `Display` node in the `Reference Manual`.
-3. If you want add 3d graphics to the main window, this can be done by using the `Supervisor` API.
-The `Supervisor` API can be used to create new nodes - meaning that you can create an IndexedFaceSet or IndexedLineSet and adjust vertex positions or indexing accordingly to create and update the shape you wish to draw in the 3D scene.
+Please lookup for the [Display](../reference/display.md) node in the `Reference Manual`.
```
Comment (stefaniapedrazzi) [inline guide/programming.md]: Reference Manual could link to `../reference/index.md` (same just above)

File: guide/the-user-interface.md [hunk 5]
```diff
@@ -47,18 +47,18 @@ In order to reset the simulation, the following steps are performed:
   - The random seeds used by Webots internally are reset.
   - All the nodes are reset. This has the following implication depending on the node type:
     - **Brake**: The brake is released.
-    - **Charger**: The `battery` field and the `emissiveColor` field of the Material node of the first Shape child node are restored.
+    - **Charger**: The `battery` field and the `emissiveColor` field of the [Material](../reference/material.md) node of the first [Shape](../reference/shape.md) child node are restored.
```
Comment (stefaniapedrazzi) [inline guide/the-user-interface.md]: is not possible to add a link to the node written in bold too?

File: guide/epuck.md (hunk 2 above)
Comment (DavidMansolino) [reply guide/epuck.md]: You right I had an (almost) complete search but only with capital.

File: guide/the-user-interface.md (hunk 5 above)
Comment (DavidMansolino) [reply guide/the-user-interface.md]: I did not put them as link because I was afraid it will remove the bold, but maybe it is not the case, I will try.

**Comment by DavidMansolino** [comment]
Thank you for your comments, they are all very pertinent.

> Here are a couple of comments, but there are very likely many others.

Yes, of course, this is just a first pass but I think that even if we are missing some links (I focused the search on the capitalized device name) it will already be significantly better.

File: guide/the-user-interface.md (hunk 5 above)
Comment (DavidMansolino) [reply guide/the-user-interface.md]: It seems to work fine:
capture du 2018-06-21 16-41-23
I will change all of them to links.

**Comment by DavidMansolino** [comment]
:+1:

File: guide/cpp-java-python.md [hunk 6]
```diff
@@ -28,15 +28,14 @@ Usually the C++/Java/Python methods have the same parameters as their C API coun
...
-If in the Scene Tree a robot is of type [Supervisor](../reference/supervisor.md), then the C++/Java/Python controller class must be derived from the `Supervisor` class, etc.
+For example, if in the Scene Tree a robot is of type [Supervisor](../reference/supervisor.md), then the corresponding C++/Java/Python controller class must extend the `Supervisor` class.
 
-As you can see in [this figure](#a-small-subset-of-webots-oriented-object-apis), both `DifferentialWheels` and `Supervisor` are subclasses of the `Robot` class.
-Hence it is possible to call the `Robot`'s methods, such as, e.g., the `step` or `getLED` functions, from the `DifferentialWheels` and `Supervisor` controllers.
-But it is not possible to call the `Supervisor` methods from a `DifferentialWheels` controller, and vice versa.
-For example it won't be possible to call the `worldReload` function from a `DifferentialWheels` controller.
+As you can see in [this figure](#a-small-subset-of-webots-oriented-object-apis), `Supervisor` is subclasses of the `Robot` class.
```
Comment (stefaniapedrazzi) [inline guide/cpp-java-python.md]: is subclasses -> is a subclass

File: guide/programming.md [hunk 7]
```diff
@@ -40,12 +40,12 @@ There are also some functions (see below) that can be used to get the velocity d
...
 
 To get the angular velocity of a robot (or robot part) in the robot's controller code: add a [Gyro](../reference/gyro.md) node to the robot (or robot part), then use the `wb_robot_get_device`, `wb_gyro_enable` and `wb_gyro_get_values` functions.
-You will find more information about the [Gyro](../reference/gyro.md) node and functions in the `Reference Manual`.
+You will find more information about the [Gyro](../reference/gyro.md) node and functions in the [Reference Manual](../reference/gyro.md).
 
 #### Get Velocity in Supervisor:
 
 Using the `wb_supervisor_node_get_velocity` function it is possible to retrieve both the linear and angular velocity of any [Solid](../reference/solid.md) node.
-You will find more information about this function in the `Reference Manual`.
+You will find more information about this function in the [Reference Manual](../reference/solid.md).
```
Comment (stefaniapedrazzi) [inline guide/programming.md]: shouldn't refer to the supervisor page instead of the solid one?

**Comment by DavidMansolino** [comment]
:+1:

File: guide/controller-plugin.md [hunk 8]
```diff
@@ -31,13 +31,13 @@ After its loading, some controller plugin functions (entry points) are called by
 A set of entry points have to be defined to let the controller plugin work smoothly.
 Some of these entry points are required and some are optional.
 
-The `Robot` node defines the location of the controller plugin through its *window* and its *remoteControl* fields (cf. reference manual).
+The [Robot](../reference/robot.md) node defines the location of the controller plugin through its *window* and its *remoteControl* fields (cf. [Reference Manual](../reference/index.md)).
```
Comment (omichel) [inline guide/controller-plugin.md]: These links to the reference manual are a little bit useless. I would either remove them or link to the robot node of the reference manual.

File: guide/controller-plugin.md [hunk 9]
```diff
@@ -188,7 +188,7 @@ A pose is defined by a set of commands (in position) of the robot motors.
 
 The motion is stored in a motion file in a user-readable format.
 The controller API allows to play easily the motion files on your robot.
-You can get the complete motion API in the reference manual.
+You can get the complete motion API in the [Reference Manual](../reference/index.md).
```
Comment (omichel) [inline guide/controller-plugin.md]: We should link to the motion API here.

File: guide/controller-plugin.md (hunk 8 above)
Comment (DavidMansolino) [reply guide/controller-plugin.md]: Yes I think we can remove it as 'Robot' already points to the reference manual.

**Comment by DavidMansolino** [comment]
:+1:

File: guide/controller-plugin.md [hunk 10]
```diff
@@ -129,15 +129,15 @@ The entry points of a native robot window controller plugin are:
...
+    *showWindow* field of the [Robot](../reference/robot.md) node is enabled.
 
 - `void *wbw_robot_window_custom_function(void *)`
 
     This function can optionally be defined to communicate and pass data from and to
     the controller program. It is called directly by the user from the controller
     when executing the `wb_robot_window_custom_function` C function or
     `Robot::windowCustomFunction` C++ function. You can find more information
-    in the reference manual.
+    in the [Reference Manual](../reference/index.md).
```
Comment (omichel) [inline guide/controller-plugin.md]: We should probably refer to ../reference/robot.md here.

File: guide/tutorial-5-compound-solid-and-physics-attributes-15-minutes.md [hunk 11]
```diff
@@ -138,20 +138,20 @@ We recommend values between *8* and *16* for regular use of Webots.
...
 The default values are well defined for regular use of Webots.
-We recommend to read the `Reference Manual` and the documentation of [ODE](http://ode-wiki.org/wiki/index.php?title=Manual) (physics engine used in Webots) to understand completely their purpose.
+We recommend to read the [Reference Manual](../reference/worldinfo.md) and the documentation of [ODE](http://ode-wiki.org/wiki/index.php?title=Manual) (physics engine used in Webots) to understand completely their purpose.
 
 ### Minor Physics Parameters
 
 There are also other physics parameters which are less useful in a regular use of Webots.
-A complete description of these parameters can be found in the `Reference Manual`.
-Remark simply that the Physics, WorldInfo and ContactProperties nodes contains other fields.
+A complete description of these parameters can be found in the [Reference Manual](../reference/index.md).
```
Comment (omichel) [inline guide/tutorial-5-compound-solid-and-physics-attributes-15-minutes.md]: I am not sure about what parameters we are speaking about here... but the link to the reference manual is too vague, we should be more precise, like ../reference/worldinfo.md if we mean WorldInfo parameters. Probably this sentence should be improved to bring more precise and useful information.

Comment (DavidMansolino) [reply guide/tutorial-5-compound-solid-and-physics-attributes-15-minutes.md]: I thought that this was part of the exercise to let the user find the information by himself and at the same time learn how to browse the reference manual. 
And the precise information and links are given in the next line:
> Remark simply that the [Physics](../reference/physics.md), [WorldInfo](../reference/worldinfo.md) and [ContactProperties](../reference/contactproperties.md) nodes contains other fields.

**Comment by DavidMansolino** [comment]
:+1:

**Pull request merged**